docs: Jaustin/goose-allowlist (#2345)
block/goose: documentation/docs/guides/allowlist.md

Reviewer: for my own understanding, does the allowlist block installation or only usage of tools?

Author: "registration" which seems to mean adding the Extension itself.  As I understand, if you can't register an extension, then you can't use any of the tools?

Reviewer: ok "installation". there's no where in the docs/app do we use "registration" as a term.

---
sidebar_position: 19
title: Goose Extension Allowlist
sidebar_label: Goose Extension Allowlist
---

> This guide explains how Goose controls which extensions you can use. Your organization creates an approved list of safe extensions that work with both the desktop app and command line version of Goose. This keeps your system secure by making sure only trusted extensions can run commands. The approved list is also known as an "allowlist".

Reviewer: ```suggestion
Goose is an extensible framework that, by default, allows you to install any MCP server. However when using Goose in a corporate setting, you may want stricter controls on which MCP servers can be installed as extensions. This guide explains how your organization can create an approved list of safe extensions that work with both the desktop app and command line version of Goose. This keeps your system secure by making sure only trusted extensions can run commands. The approved list is known as an **allowlist**.
```

## Overview

> An allowlist lets administrators control which commands can run in Goose's Stdio extensions. This security feature stops harmful extensions from running unapproved commands on your computer. When turned on, the Goose server agent only allows commands that are on the list. Any command not on this list are blocked, and a Goose user see an error message.

Reviewer: ```suggestion
An allowlist lets administrators control which commands can run in Goose's `stdio` extensions. This security feature stops harmful extensions from running unapproved commands on your computer. When turned on, the Goose server agent only allows commands that are on the list. Any command not on this list is blocked, and the user see an error message.
```

## How It Works

1. Goose fetches the allowlist from a URL specified by the `GOOSE_ALLOWLIST` environment variable.
2. The allowlist is a YAML file that contains a list of allowed extension commands.
3. The allowlist is fetched once when first needed and cached for the lifetime of the Goose server.

> 4. When a Stdio extension is [added](/docs/getting-started/using-extensions#adding-extensions), Goose checks the command against the allowlist.

Reviewer: ```suggestion
4. When an extension is [added](/docs/getting-started/using-extensions#adding-extensions), Goose checks the MCP server's installation command against the allowlist.
```

ah by command, you mean the installation command. this was confusing me but i got it now

> 5. If the command is not in the allowlist, the extension registration is rejected.

Reviewer: ```suggestion
5. If the command is not in the allowlist, the extension installation is rejected.
```

---
sidebar_position: 19
title: Goose Extension Allowlist
sidebar_label: Extension Allowlist
---

Goose is an extensible framework that, by default, allows you to install any MCP server. However, you may want stricter controls on which MCP servers can be installed as extensions (e.g. in a corporate setting). 

This guide explains how you can create an **allowlist** of safe extensions that work with Goose Desktop and CLI. An allowlist lets administrators control which MCP servers can be installed as Goose extensions. When enabled, Goose will only install extensions that are on the list, and will block installation of any others.

## How It Works

1. The allowlist is a YAML file that contains a list of allowed extension commands.
2. Goose fetches the allowlist from a URL specified by the `GOOSE_ALLOWLIST` environment variable.

> 3. The allowlist is fetched when first needed and is cached.

Reviewer: if the allowlist is only fetched once and then cached, how does goose know when/if it's been updated?

5. If the command is not in the allowlist, the extension installation is rejected.

## Configuration

### 1. Create and Deploy Allowlist

The allowlist must be a YAML file with the following structure:

```yaml
extensions:
  - id: extension-id-1
    command: command-name-1
  - id: extension-id-2
    command: command-name-2
  # ... more extensions
```

#### Example

In this example, only the Slack, GitHub, and Jira extensions can be installed: 

```yaml
extensions:
  - id: slack
    command: uvx mcp_slack
  - id: github
    command: uvx mcp_github
  - id: jira
    command: uvx mcp_jira
```

After creating the allowlist, you must deploy it to a URL.

### 2. Set Environment Variable

Create an environment variable called `GOOSE_ALLOWLIST` and set the value to the URL of your YAML file:

```bash
export GOOSE_ALLOWLIST=https://example.com/goose-allowlist.yaml
```

You can also add this export to your shell configuration file (On a Mac, it's your `~/.bashrc` or `~/.zshrc` file). 

:::info
If this environment variable is not set, no allowlist restrictions are applied. With no restrictions, all extensions can be installed.
:::

> ## Best Practices for Defining Allowlist Entries

Reviewer: ```suggestion
## Best Practices
```

To effectively use the allowlist with exact matching:

1. **Specify commands**: Define the exact command string that you want to allow.
2. **Include full paths**: If you want to allow a command only from a specific path, include the full path in the allowlist.
3. **Audit regularly**: Review your allowlist frequently to ensure it only contains the commands you intend to allow.

> 4. **Use HTTPS**: Always use HTTPS URLs in your allowlist to prevent man-in-the-middle attacks.

Reviewer: ```suggestion
4. **Use HTTPS**: Use an HTTPS URL for your allowlist to prevent man-in-the-middle attacks.
```
